### ISSUE TITLE
Keine Logfiles in var/logs

### ISSUE THREAD

**Issue by wschern** [contao/contao#1527]
**Affected version(s)**

Contao 4.9.1

**Description**

Ich bekomme eine Fehlerseite in Contao "Es ist ein Fehler aufgetreten
", aber es wird keine 'prod' Log-Datei unter var/logs erstellt. Auch die Serverlogs sind leer. Mein Hoster ist Hetzner.
P.S. dev-... Logs werden generiert, nur die prod-... Logs nicht.

**Comment by cmartin6** [comment]
Ich kann das ebenso unter Contao 4.9.1 reproduzieren. Keine Logs unter var/logs - Aktuell sehr ärgerlich...

**Comment by chuckki** [comment]
Lassen Sie uns den Entwicklern zusammen helfen und das beschriebene Verhalten genauer anschauen; könnte jemand vielleicht sagen, welche|r Fehler nun tatsächlich auftrat|en? Die dev-Logs funktionieren anscheinend und deren Inhalt könnte Aufschluss geben. @cmartin6 Sie konnten den Fehler reproduzieren, welche Vorraussetzungen waren dafür nötig?

**Comment by cmartin6** [comment]
Mittlerweile sehe ich, dass in einer meiner Contao 4.9.1 Instanzen auch prod-Logs generiert wurden. Was dabei aber auffällig ist, ist dass sämtliche PHP Fehlermeldungen (404, 500, etc.) nicht darin aufscheinen. 
Grundsätzlich kann ich keine genauen Vorraussetzungen zur Reproduktion nennen: Ich habe eine frische Contao 4.9.1 Instanz aufgesetzt, die prod-Logs verhalten sich jedoch nicht mehr so, wie es beispielsweise unter Contao 4.4.47 der Fall war...

**Comment by chuckki** [comment]
40x werden per default nicht in die prod-Logs geschrieben siehe https://github.com/contao/contao/blob/50a3027d47e6ac1bae3b3e7a139be512a9c36cc5/manager-bundle/src/Resources/skeleton/config/config_prod.yml#L18 
500er Fehler sollten in den prod-Logs vorhanden sein.

**Comment by leofeyer** [comment]
@wschern: Hat @chuckki's Erklärung Deine Frage beantwortet? Dann schließe das Ticket bitte.

**Comment by fritzmg** [comment]
I have actually observed this as well, as did someone else on Slack and [on the forum](https://community.contao.org/de/showthread.php?76125-wo-befindet-sich-die-error-log-datei). When looking into it, I noticed that the `fingers_crossed` logger is buffering the log entries, i.e. it does not write the log immediately. But I couldn't figure out the exact logic behind it.

**Comment by plorenz-etes** [comment]
I also experience this issue in various installations. While some seem to work fine, others don't write prod log files at all (only in debug mode, dev log files will be written). It is also not dependent on the error code - even 50x error codes don't create log entries. In Contao 4.4 this was working fine.

**Comment by leofeyer** [comment]
As discussed in Mumble on August 27th, we would need to debug this in an installation where the problem occurs.

**Comment by RoflCopter24** [comment]
I have a similar problem, where there are __no__ Exceptions logged at all, only debug and info messages

What I noticed is, that in my case there is never an EventListener for the event "kernel.exception" registered, that actually writes an exception to the Logger (you can see them using `vendor/bin/contao-console debug:event-dispatcher kernel.exception`)

In a Contao 4.4.* and a 4.6 version I tested, there is a Listener registered that writes the Exception to the Logger:
` #6      Symfony\Component\HttpKernel\EventListener\ExceptionListener::onKernelException()   -128`

In my Contao 4.9.* installation, the Symfony-Listener is missing, there should be a line
`Symfony\Component\HttpKernel\EventListener\ErrorListener` 
(ExceptionListener was deprecated in the newer Symfony version) but there is none (see [Contao_4_9_EventListeners.txt](https://github.com/contao/contao/files/5157252/Contao_4_9_EventListeners.txt)).
If I add one myself, then Exceptions get written to the logs as expected.
The Installation was installed using the CommandLine from a predefined composer.json

__However__ if I install a blank 4.9 Contao using the contao manager, everything works as expected and the ExceptionListener is there...

I will do some more research on this intriguing matter and see if there is a bundle or custom configuration that causes this.

**Comment by aschempp** [comment]
> The Installation was installed using the CommandLine from a predefined composer.json

Does that mean you're not using the Contao Managed Edition (`contao/manager-bundle`)?

**Comment by RoflCopter24** [comment]
> Does that mean you're not using the Contao Managed Edition (contao/manager-bundle)?

No, sorry that was probably not clear enough. I was trying to say that it was not freshly installed using the Contao Manager. The composer.json is based on the Managed Edition but has been copied from another project and modified, then installed using the command line :-)

Unfortunately I did not have time to pursue this issue further, so no news from me just yet.

For reference, here is the composer.json used:
```json
{
    "type": "project",
    "repositories": [
        {
            "type": "composer",
            "url": "https://rapid-data.gitlab.io/"
        },
        {
            "type": "git",
            "url": "https://github.com/RoflCopter24/contao-rocksolid-icon-picker"
        },
        {
            "type": "path",
            "url": "../bundles/funeral-configurator-bundle"
        }
    ],
    "require": {
        "php": "^7.3",

        "contao/faq-bundle": "4.9.*",
        "contao/manager-bundle": "4.9.*",
        "contao/news-bundle": "4.9.*",
        "contao/conflicts": "@dev",

        "bugbuster/contao-cron-bundle": "^1.3",
        "contao-bootstrap/bundle": "^2.1",
        "delahaye/dlh_googlemaps": "^2.4",
        "fipps/contao-rssimport-bundle": "^1.2",
        "jrgregory/m17-sticky-backend-footer": "^2.0",
        "madeyourday/contao-rocksolid-custom-elements": "^2.2",

        "rapid-data/funeral-configurator-bundle": "*",

        "rsm/rsm-googlerecaptcha-modul": "^1.0",
        "christianbarkowsky/rel-canonical": "^3.3",
        "hofff/contao-social-tags": "^1.0",

        "terminal42/contao-ajaxform": "^1.0",
        "terminal42/notification_center": "^1.4",
        "terminal42/contao-leads": "^1.4",
        "terminal42/contao-conditionalformfields": "^2.2",
        "terminal42/contao-easy_themes": "^2.2",
        "terminal42/contao-mp_forms": "^3.0",

        "smart-core/accelerator-cache-bundle": "^1.2",
        "doctrine/doctrine-migrations-bundle": "^1.2",
        "mashape/unirest-php": "^3.0",
        "roflcopter24/contao-rocksolid-icon-picker": "^2.1"
    },
    "require-dev": {
        "deployer/deployer": "^6.4",
        "deployer/recipes": "^6.2",
        "symfony/maker-bundle": "^1.14",
        "symfony/phpunit-bridge": "^5.0",
        "phpunit/phpunit": "^8.4",
        "doctrine/doctrine-fixtures-bundle": "^3.3"
    },
    "minimum-stability": "dev",
    "prefer-stable": true,
    "autoload": {
        "psr-4": { "App\\": "src/" }
    },
    "extra": {
        "contao-component-dir": "assets"
    },
    "scripts": {
        "post-install-cmd": [
            "Contao\\ManagerBundle\\Composer\\ScriptHandler::initializeApplication"
        ],
        "post-update-cmd": [
            "Contao\\ManagerBundle\\Composer\\ScriptHandler::initializeApplication"
        ]
    }
}
```

**Comment by RoflCopter24** [comment]
__An update__: I got finally some time to go through all the extensions and remove them step by step. 

It revealed that the Exception listener vanishes from the Symfony system when I install the `"contao-bootstrap/bundle"` bundle.
So if anyone experiences this problem, check if you use the contao-bootstrap bundle and try if removing it fixes the issue.

These are the registered event listeners __before__ installing the bundle:
![image](https://user-images.githubusercontent.com/8280334/94539911-2401dc00-0246-11eb-9a53-0fd0edb9d5af.png)

And this is the same list __after__ installing `contao-bootstrap/bundle`:
![image](https://user-images.githubusercontent.com/8280334/94540946-52cc8200-0247-11eb-9b0e-1a1c1fcbd27a.png)

Notice how the Exception-Listener including the call `ExceptionListener::logKernelException` and `ExceptionListener::onKernelException` are missing.

-> Therefore I conclude that the issue lies not in the Contao-Core but in one of these bundles:
```
  - Installing netzmacht/html (2.0.1): Loading from cache
  - Installing symfony/templating (v4.4.14): Loading from cache
  - Installing contao-community-alliance/meta-palettes (2.0.7): Loading from cache
  - Installing netzmacht/contao-form-designer (1.1.6): Loading from cache
  - Installing contao-bootstrap/core (2.0.3): Loading from cache
  - Installing contao-bootstrap/form (2.1.5): Loading from cache
  - Installing contao-bootstrap/templates (2.1.0): Loading from cache
  - Installing netzmacht/contao-toolkit (3.4.5): Loading from cache
  - Installing contao-bootstrap/tab (2.0.4): Loading from cache
  - Installing contao-bootstrap/panel (2.1.2): Loading from cache
  - Installing contao-bootstrap/grid (2.2.0): Loading from cache
  - Installing contao-bootstrap/navbar (2.1.4): Loading from cache
  - Installing contao-bootstrap/layout (2.0.4): Loading from cache
  - Installing contao-bootstrap/bundle (2.1.1)
```

**Comment by dmolineus** [comment]
@RoflCopter24 Does happen a downgrade for any installed symfony component when installing `contao-bootstrap/bundle`?

**Comment by dmolineus** [comment]
This is what happens:

1. `contao-bootstrap` requires `netzmacht/contao-toolkit`
2. `netzmacht/contao-toolkit` provides a templating engine `toolkit` which abstracts Contao template handling
3. The result is that templating is enabled in the framework bundle with the only registered engine `toolkit`
4. However the twig-bundle removes the `exception_listener` now, see             https://github.com/symfony/symfony/blob/72ff4012a533be78b5ee77f3ace8606937e9b32f/src/Symfony/Bundle/TwigBundle/DependencyInjection/Compiler/ExceptionListenerPass.php#L37

If I really understand what the bundle want to do is to remove the exception listener **if** twig is used as engine. So the implementation is wrong. If it make sense to kill the exception listener at all, is another question. Having twig enabled does not mean it is used...

To summarize the usage of `contao-bootstrap` requires `netzmacht/contao-toolkit` and this triggers a bug in the symfony twig bundle.

**Comment by Aybee** [comment]
Maybe for other reasons, but just this week I had the same case in another Symfony installation. No log files are written.
https://github.com/kevinpapst/kimai2/blob/master/composer.json

**Comment by leofeyer** [comment]
@dmolineus Thank you for debugging this. I am closing this issue then.

**Comment by dmolineus** [comment]
Fyi, my proposed solution got accepted and merged and so will be shipped with the next symfony 4.4.x release.
